src/pages/App.tsx

```diff
@@ -106,8 +110,33 @@ export default function App() {
     }
   ]
 
+  const [walletModalActive, setWalletModalActive] = React.useState<boolean>(false)
+
+  const handleMessengerStatus = useCallback(() => {
+    setWalletModalActive(!walletModalActive)
+  }, [walletModalActive, setWalletModalActive])
+
+  const { account } = useWeb3React<Web3Provider>()
+
   return (
     <Suspense fallback={null}>
+      <Intercom
+        appID={'uj2kwt4z'}
+        walletAddress={account as string}
+        onMessengerOpen={handleMessengerStatus}
+        onMessengerHide={handleMessengerStatus}
+      />
+      {!account && (
+        <WalletModal
+          open={walletModalActive}
+          closeModal={() => {
+            setWalletModalActive(false)
+          }}
+          onWalletConnect={wallet => {
+            console.log(wallet)
+          }}
+        />
+      )}
```

Comment:
Any particular reason, why do we need to integrate WalletModal here ? We already have wallet modal when user click on "Connect Wallet" in header. I think we should not ask user forcefully to connect wallet.

Reply:
@bilalcorbacioglu sounds good but you are also asking user to connect on `onHide`, is that expected ?